lib/spack/spack/environment/environment.py

```diff
@@ -1535,7 +1535,7 @@ def _get_specs_to_concretize(
 
     def _concretize_together_where_possible(
         self, tests: bool = False
-    ) -> List[Tuple[spack.spec.Spec, spack.spec.Spec]]:
+    ) -> Sequence[Tuple[spack.spec.Spec, spack.spec.Spec]]:
```

Comment:
If we have `SpecPair` in scope, we should use it here. If we don't, we should either import it or we should stop using it elsewhere in this file.

Reply:
Done in https://github.com/spack/spack/pull/47382/commits/72d2cc420f07a7edbb091ca0941ce2644a427dc3